extensions/amp-story-shopping/0.1/amp-story-shopping-attachment.js

```diff
@@ -162,14 +166,18 @@ export class AmpStoryShoppingAttachment extends AMP.BaseElement {
    * @private
    */
   updateTemplate_(shoppingData) {
-    const shoppingDataForPage = this.shoppingTags_.map(
-      (shoppingTag) => shoppingData[shoppingTag.getAttribute('data-product-id')]
+    const pageElement = closestAncestorElementBySelector(
+      this.element,
+      'amp-story-page'
     );
 
     let productForPdp = shoppingData.activeProductData;
     // If no active product and only one product on page, use the one product for the PDP.
-    if (!shoppingData.activeProductData && shoppingDataForPage.length === 1) {
-      productForPdp = shoppingDataForPage[0];
+    if (
+      !shoppingData.activeProductData &&
+      shoppingData[pageElement.id].length === 1
+    ) {
+      productForPdp = shoppingData[pageElement.id][0];
```

Comment:
Driving by with a minor suggestion about structure. The function might be better defined if it matches the scope of the data. Make `shoppingData` equivalent to the page's data itself, like before, and pass the actual field value `shoppingData[pageElement.id]` to this function.